src/main/java/com/aws/greengrass/disk/spool/DiskSpoolDAO.java

```diff
@@ -136,63 +142,65 @@ public synchronized SpoolMessage getSpoolMessageById(long messageId) throws SQLE
      * @throws SQLException when fails to insert SpoolMessage in the database
      */
     @SuppressWarnings({"PMD.ExceptionAsFlowControl", "PMD.AvoidCatchingGenericException"})
-    public synchronized void insertSpoolMessage(SpoolMessage message) throws SQLException {
-        String sqlString =
-                "INSERT INTO spooler (message_id, retried, topic, qos, retain, payload, userProperties, "
-                        + "messageExpiryIntervalSeconds, correlationData, responseTopic, payloadFormat, contentType) "
-                        + "VALUES (?,?,?,?,?,?,?,?,?,?,?,?);";
-        Publish request = message.getRequest();
-        try (PreparedStatement pstmt = dbConnection.prepareStatement(sqlString)) {
-            pstmt.setLong(1, message.getId());
-            pstmt.setInt(2, message.getRetried().get());
+    public void insertSpoolMessage(SpoolMessage message) throws SQLException {
+        try (LockScope ignored = LockScope.lock(dbConnectionLock.writeLock())) {
```

Comment:
Like I said in my comment, use the `readLock` for all usage of the connection. We're not trying to synchronize database access. We're trying to synchronize modifying `connection`. So always use `readLock`, except when modifying `connection` itself.

Reply:
+1, looks like this still needs to be updated